private/protocol/query/queryutil/queryutil.go

```diff
@@ -3,6 +3,7 @@ package queryutil
 import (
```

Comment:
is there a reason that the `query` implementation doesnt need any accompanied unit tests?